### PR TITLE
librados, osd: clang fixes

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

**Comment by tchaikov** [comment]
address some issues reported at http://people.redhat.com/bhubbard/scan-build.17-03-02/index.html

**Comment by badone** [comment]
Please test this

File: src/osd/PGLog.h [hunk 1]
```diff
@@ -1115,10 +1115,10 @@ struct PGLog : DoutPrefixProvider {
...
     missing_type &missing, ostringstream &oss,
-    bool *clear_divergent_priors = NULL,
-    const DoutPrefixProvider *dpp = NULL,
-    set<string> *log_keys_debug = 0,
-    bool debug_verify_stored_missing = false
+    bool *clear_divergent_priors,
+    const DoutPrefixProvider *dpp,
+    set<string> *log_keys_debug,
+    bool debug_verify_stored_missing
     ) {
```
Comment (badone) [inline src/osd/PGLog.h]: The call site at ceph_objectstore_tool.cc:331 (there mat be others) relies on the defaults being set.

**Comment by tchaikov** [comment]
@badone fixed & repushed

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2017-03-06_11:46:50-rados-wip-kefu-testing---basic-mira/

**Pull request merged**